src/client_side_request.cc

```diff
@@ -715,9 +715,10 @@ ClientRequestContext::clientAccessCheckDone(const Acl::Answer &answer)
     acl_checklist = nullptr;
     err_type page_id;
     Http::StatusCode status;
+    const SBuf none("[none]");
     debugs(85, 2, "The request " << http->request->method << ' ' <<
            http->uri << " is " << answer <<
-           "; last ACL checked: " << (AclMatchedName ? AclMatchedName : "[none]"));
+           "; last ACL checked: " << AclMatchedName.value_or(none));
```

Comment:
Please encapsulate this duplicated reporting logic as

```suggestion
           "; last ACL checked: " << answer.lastCheckDescription());
```

Acl::Answer::lastCheckDescription() should use a _static_ SBuf local for returning the "[no-ACL]" value reference (when Acl::Answer::lastCheckedName is missing).

Reply:
Done.